onmt/utils/optimizers.py

```diff
@@ -259,7 +259,8 @@ def from_opt(cls, model, opt, checkpoint=None):
         optim_opt = opt
         optim_state_dict = None
 
-        if opt.train_from and checkpoint is not None:
+        if (opt.train_from and checkpoint is not None and
+                'optim' in checkpoint.keys()):
```

Comment:
I'm not sure this is a good idea. We could technically want to run this code with a released checkpoint (model['optim'] = None) and reset_optim = all.

Reply:
that's the point, you can't. The typical use case was nllb-200 checkpoint with no optimizer in it. it has to recreate the optimizer from scratch.

Reply:
to be more specific.
case 'optim' key does not exist: we need to create the optimizer from scratch and the previous codebase would spit an error.
case 'optim' exist but checkpoint['optim'] = None: it goes through the code as it was.

Reply:
Makes sense, read it the other way around;